app/calculation/index.js

```diff
@@ -2,15 +2,56 @@ const getPaymentBands = require('./get-payment-band')
 const calculateBand = require('./calculate-band')
 const calculateOverall = require('./calculate-overall')
 const { convertStringToPence } = require('./convert-currency')
+const schemeYears = require('./scheme-years')
+
+function transformResults (bandResults) {
+  return Object.values(bandResults.flat().reduce((r, { band, result }) => {
+    if (!r[band]) {
+      r[band] = {
+        band: band,
+        result: []
+      }
+    }
+    result.map((bandResult) => {
+      r[band].result.push(bandResult)
+      return bandResult
+    })
+    return r
+  }, {}))
+}
+
+function calculate (schemeYearValues) {
+  const bandResults = []
+
+  schemeYearValues.forEach(schemeYearValue => {
+    const bpsValueInPence = convertStringToPence(schemeYearValue.bpsValue.toString())
+    const paymentBands = getPaymentBands(schemeYearValue.schemeYear, bpsValueInPence)
+    const bandResult = paymentBands.map(calculateBand)
+    bandResults.push(bandResult.flat())
+  })
+
+  return transformResults(bandResults)
+}
+
+function calculateSingle (bpsValue) {
+  const schemeYearValues = []
+  schemeYears.forEach((schemeYear) => schemeYearValues.push({ schemeYear, bpsValue }))
```

Comment:
You could simplify this by using `map` rather than `foreach`

ie no need to declare an empty array and push to it, you can just use map to return a new array.  If you want each item in the original array to map to a new object then map is ideal

```
schemeYearValues = schemeYears.map(schemeYear => ({ schemeYear, bpsValue }))
```